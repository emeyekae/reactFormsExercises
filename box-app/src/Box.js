import React from "react";

function Box({
    id, handleRemove, height=300, width=300,  backgroundColor="purple"
}) 
{ 
    const remove = () => handleRemove(id);
    return (
        <div>
            <div style={{width: `${width}px`, height: `${height}px`, backgroundColor }}/>
            <button onClick={remove}>X</button>
        </div>
    );
}

export default Box;