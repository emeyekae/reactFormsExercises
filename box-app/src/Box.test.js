import React from "react";
import { render, screen } from '@testing-library/react';
import Box from './Box';

//smoke test
test('renders without crashing', function() {
  render(<Box />);
});

//snapshot test
test("matches snapshot", function() {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});