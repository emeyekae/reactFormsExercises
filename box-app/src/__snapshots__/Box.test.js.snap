// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="width: 300px; height: 300px; background-color: purple;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
